doc/api/stream.md

```diff
@@ -154,89 +325,134 @@ Examples of readable streams include:
+[`EventEmitter`][] interface.
+
+In paused mode, the [`stream.read()`][stream-read] method must be called
+explicitly to read chunks of data from the stream.
+
+All [Readable][] streams begin in paused mode but can be switched to flowing
+mode in one of the following ways:
+
+* Adding a [`'data'`][] event handler.
+* Calling the [`stream.resume()`][stream-resume] method.
+* Calling the [`stream.pipe()`][] method to send the data to a [Writable][].
+
+The Readable can switch back to paused mode using one of the following:
+
+* If there are no pipe destinations, by calling the
+  [`stream.pause()`][stream-pause] method.
+* If there are pipe destinations, by removing any [`'data'`][] event
+  handlers, and removing all pipe destinations by calling the
+  [`stream.unpipe()`][] method.
+
+*Note*: For backwards compatibility reasons, removing [`'data'`][] event
+handlers will **not** automatically pause the stream. Also, if there are piped
+destinations, then calling [`stream.pause()`][stream-pause] will not guarantee
+that the stream will *remain* paused once those destinations drain and ask for
+more data.
 
-Emitted when the stream and any of its underlying resources (a file
-descriptor, for example) have been closed. The event indicates that
-no more events will be emitted, and no further computation will occur.
+*Note*: If a [Readable][] is switched into flowing mode and there are no event
+handlers or [`stream.pipe()`] destinations attached to handle the data, that
+data will be lost.
 
-Not all streams will emit the `'close'` event as the `'close'` event is
-optional.
+#### Class: stream.Readable
+
+<!--type=class-->
 
-#### Event: 'data'
+##### Event: 'close'
+
+The `'close'` event is emitted when the stream and any of its underlying
+resources (a file descriptor, for example) have been closed. The event indicates
+that no more events will be emitted, and no further computation will occur.
+
+Not all [Readable][] streams will emit the `'close'` event.
+
+##### Event: 'data'
 
 * `chunk` {Buffer|String} The chunk of data.
 
-Attaching a `'data'` event listener to a stream that has not been
-explicitly paused will switch the stream into flowing mode. Data will
-then be passed as soon as it is available.
+The `'data'` event is emitted when the stream is in flowing mode and there is
+data available to be consumed. Attaching a `'data'` event listener to a stream 
+that has not been explicitly paused will switch the stream into flowing mode. 
+Data will then be passed as soon as it is available.
 
-If you just want to get all the data out of the stream as fast as
-possible, this is the best way to do so.
+The listener callback will be passed the chunk of data as a string if a default
+encoding has been specified for the stream using the 
+`readable.setEncoding()` method; otherwise the data will be passed as a
+`Buffer`.
+
+The `'data'` event is the most efficient means of consuming all of a Readable
+streams data:
```

Comment:
Efficient in what terms? (Notably the "as fast as possible" suffers from the same problems, but we should be clear that the example below optimizes for parsimony of implementation)